task_schedule.pl

```diff
@@ -196,9 +197,20 @@
     # Update heartbeat file
     eval { 	io($opt{heartbeat})->write(qq{Running on $hostname}); };
     if ($@) {
-		heartbeat_perm_fail($@);
+		email_and_die($@, "ERROR - Cannot update heartbeat");
     }
 
+    eval{
+        my $cwd = getcwd();
+        if ($cwd eq ''){
```

Comment:
I wasn't sure about the ways this can go badly, so here I've called $cwd equal to '' an error but also handled any other exception on getcwd().